File: Model-View-ViewModel-Java/app/src/main/java/com/mvvm/java/sample/login/LoginActivity.java
package com.mvvm.java.sample.login;

import com.mvvm.java.core.base.mvvm.MVVMActivity;
import com.mvvm.java.core.di.component.ActivityComponent;
import com.mvvm.java.databinding.ActivityLoginBinding;

public class LoginActivity extends MVVMActivity<ActivityLoginBinding, LoginViewModel> {
    @Override
    public ActivityLoginBinding createViewBinding() {
        ActivityComponent component = getActivityComponent();
        if (component != null) {
            component.inject(this);
        }
        return ActivityLoginBinding.inflate(getLayoutInflater());
    }

    @Override
    public void setup() {

    }
}

File: Model-View-ViewModel-Kotlin/app/src/main/java/com/base/mvvm/samples/ui/login/LoginActivity.kt
package com.base.mvvm.samples.ui.login

import androidx.activity.viewModels
import com.base.mvvm.R
import com.base.mvvm.core.base.BaseActivityViewModel
import com.base.mvvm.databinding.ActivityLoginBinding

class LoginActivity : BaseActivityViewModel<ActivityLoginBinding, LoginViewModel>() {

    private val viewModel: LoginViewModel by viewModels()

    override val layoutId: Int
        get() = R.layout.activity_login

    override fun createViewModel(): LoginViewModel = viewModel


}